### DISCUSSION TITLE
referrer shortcuts \ presets

### DISCUSSION THREAD

**Discussion by Owyn** [iptv-org/community#144] [Ideas]
we currently have shortcuts for icons, langauge etc by using `tvg-id`, 

it would be very nice to also have such shortcuts for referrers - because there are probably just few referrers needed universally for channels to work but you have to enter them each and every time for every channel in **two** places now `user-agent=" "` and `EXTVLCOPT`

so something like `user-agent="Preset1"` where `Preset1` is defined in a separate file bot later auto-replaces from (same as with `tvg-id`s)  would be nice instead

**Comment by freearhey** [comment]
the thing is that links to streams are stored in playlists just so they can be played in any player without any changes

first of all it is needed for quick testing, but also it is a safety precaution in case the bot has to be shut down 

and for the same reason user-agent has to be duplicated

**Reply by Owyn** [reply]
the reason for duplication is solely VLC player, is it not? cuz it says there `extended VLC options` (`#EXTVLCOPT`)

> precaution in case the bot has to be shut down

well, if that happens - every modern text editor has a "search & replace" function which'd restore referrers to their full in a second everywhere

> streams are stored in playlists just so they can be played in any player without any changes

but without icons and categories and everything nice

**Reply by freearhey** [reply]
`#EXTVLCOPT` for VLC, attribute `user-agent` for other players
